File: JavaApp/src/main/java/com/charlezz/javaapp/feature/local/DBHelperImpl.java
package com.charlezz.javaapp.feature.local;

import androidx.lifecycle.LiveData;
import androidx.paging.LivePagedListBuilder;
import androidx.paging.PagedList;

public class DBHelperImpl implements DBHelper{
    private PersonDao personDao;

    public DBHelperImpl(PersonDao personDao) {
        this.personDao = personDao;
    }

    @Override
    public LiveData<PagedList<Person>> getPersons() {
        return new LivePagedListBuilder<>(
                personDao.getPersonSource(),
                new PagedList.Config.Builder()
                        .setEnablePlaceholders(false)
                        .setPageSize(20)
                        .build()
        ).build();
    }
}

File: KotlinApp/src/main/java/com/charlezz/finalarchitecture/feature/local/DBHelperImpl.kt
package com.charlezz.finalarchitecture.feature.local

import androidx.lifecycle.LiveData
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList

class DBHelperImpl (private val personDao: PersonDao) : DBHelper {
    override fun getPersons(): LiveData<PagedList<Person>> = LivePagedListBuilder(
            personDao.getPersonSource(),
            PagedList.Config.Builder()
                    .setEnablePlaceholders(false)
                    .setPageSize(20)
                    .build()).build()


}